visualization/app/codeCharta/ui/codeMap/codeMap.arrow.service.ts

```diff
@@ -110,30 +108,38 @@ export class CodeMapArrowService
 		}
 	}
 
-	private showEdgesOfBuildings(edges: Edge[]) {
-		const node = this.threeSceneService.getHighlightedNode()
-		if (this.threeSceneService.getSelectedBuilding() && node) {
-			this.buildPairingEdges(this.threeSceneService.getSelectedBuilding().node, edges)
-			this.buildPairingEdges(node, edges)
-		} else if (node) {
-			this.buildPairingEdges(node, edges)
-		} else if (this.threeSceneService.getSelectedBuilding()) {
-			this.buildPairingEdges(this.threeSceneService.getSelectedBuilding().node, edges)
-		} else {
+	private isEdgeAppliableForBuilding(codeMapBuilding: CodeMapBuilding) {
+		return this.storeService.getState().dynamicSettings.edgeMetric !== "None" && !codeMapBuilding.node.flat
+	}
+
+	private showEdgesOfBuildings(edges: Edge[], hoveredbuilding?: CodeMapBuilding) {
+		const buildings: Map<string, Node> = new Map()
+		const selectedBuilding = this.threeSceneService.getSelectedBuilding()
+
+		if (selectedBuilding) {
+			buildings.set(selectedBuilding.node.path, this.threeSceneService.getSelectedBuilding().node)
+		}
+		if (hoveredbuilding) {
+			buildings.set(hoveredbuilding.node.path, hoveredbuilding.node)
+		}
+		if (buildings.size > 0) {
+			this.buildPairingEdges(buildings, edges)
+		}
+		if (!selectedBuilding && !hoveredbuilding) {
 			this.addEdgePreview(
 				null,
 				edges.filter(x => x.visible !== EdgeVisibility.none)
 			)
 		}
```

Comment:
Move this to the top and return after that. If there is nothing hovered and nothing selected, then we don't have to check for them.